core/src/common.cpp

```diff
@@ -362,6 +364,264 @@ namespace irods::http
+
+			// Attempt to get key value parameter (JWA)
+			auto key{_jwk.get_jwk_claim("k").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::hs256(key));
+		}
+		if (_alg == "ES256") {
+			logging::trace("{}: Detected [ES256], attempting extraction of attributes from JWK...", __func__);
+
+			// Get curve parameter (JWA)
+			auto crv{_jwk.get_jwk_claim("crv").as_string()};
+
+			// Get x coordinate parameter (JWA)
+			auto x{_jwk.get_jwk_claim("x").as_string()};
+
+			// Get y coordinate parameter (JWA)
+			auto y{_jwk.get_jwk_claim("y").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::es256(jwt::helper::create_public_key_from_ec_components(crv, x, y)));
+		}
+	}
+
+	/// Bleh blah
+	///
+	/// \returns A refernce to the provided jwt::verifier, allowing for chaining.
+	auto add_algorithms_to_verifier(jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier, const jwt::jwks<jwt::traits::nlohmann_json>& _jwks, std::string_view _alg) -> jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& {
+		namespace logging = irods::http::log;
+
+		// Get the JWK the access token was signed with. This is optional.
+		// See RFC 7515 Section 4.1.4
+		try {
+			auto jwk{_jwks.get_jwk(decoded_token.get_key_id())};
+			add_alg_from_jwk(_verifier, jwk, _alg);
+		}
+		catch (const std::runtime_error& e) {
+			// We cannot pick out the specific key used...
+			// Find JWKs that match any of the following:
+			//   - 'alg' (optional) matches exactly
+			//   - 'kty' (required) matches 'alg' family [EC,RSA,oct]
+			//   - 'use' (required if sign & encrypt keys exist) to be 'sig'
+			//
+			// Additionally, symetric algoritms it /seems/ that
+			// "... the octects of the 'client_secret'..." are used for the algorithm...
+			const auto algorithm_family{_alg.substr(0,2)};
+			std::string search_string;
+
+			// RSA family
+			if (algorithm_family == "RS") {
+				search_string = "RSA";
+			}
+			// EC family
+			else if (algorithm_family == "ES") {
+				search_string = "EC";
+			}
+			// Symmetric algo
+			else if (algorithm_family == "HS") {
+				search_string = "oct";
+			}
+
+			// Go through entire key set
+			std::for_each(std::cbegin(_jwks), std::cend(_jwks), [&_verifier, &_alg, &search_string] (const auto& _jwk) -> void {
+				try {
+					if (auto key_type{_jwk.get_jwk_claim("kty").as_string()}; key_type == search_string) {
+						add_alg_from_jwk(_verifier, _jwk, _alg);
+					}
+				}
+				catch (const std::runtime_error& e) {
+					logging::error("{}: Invalid key [{}]", __func__, e.what());
+					return;
+				}
+			});
+		}
+
+		// Allow for chaining
+		return _verifier;
+	}
+	
+	/// Validates an OAuth 2.0 Access Token using
+	///
+	/// See RFC 9068 for more details on JWT for OAuth 2.0
+	/// See RFC 7515 for details on JSON Web Signature (JWS)
+	/// See RFC 7518 for details on JSON Web Algorithms (JWA)
```

Comment:
Consider using RFC XXXX [DESCRIPTION] ([SHORT HAND]), and then reference in function using [SHORT HAND].